spring/src/main/java/io/seata/spring/annotation/datasource/SeataAutoDataSourceProxyAdvice.java

```diff
@@ -0,0 +1,50 @@
+/*
+ *  Copyright 1999-2019 Seata.io Group.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package io.seata.spring.annotation.datasource;
+
+import javax.sql.DataSource;
+import java.lang.reflect.Method;
+
+import io.seata.rm.datasource.DataSourceProxy;
+import org.aopalliance.intercept.MethodInterceptor;
+import org.aopalliance.intercept.MethodInvocation;
+import org.springframework.aop.IntroductionInfo;
+import org.springframework.beans.BeanUtils;
+
+/**
+ * @author xingfudeshi@gmail.com
+ */
+public class SeataAutoDataSourceProxyAdvice implements MethodInterceptor, IntroductionInfo {
+
+    @Override
+    public Object invoke(MethodInvocation invocation) throws Throwable {
+        DataSourceProxy dataSourceProxy = DataSourceProxyHolder.get().putDataSource((DataSource) invocation.getThis());
+        Method method = invocation.getMethod();
+        Object[] args = invocation.getArguments();
+        Method m = BeanUtils.findDeclaredMethod(DataSourceProxy.class, method.getName(), method.getParameterTypes());
+        if (null != m) {
+            return m.invoke(dataSourceProxy, args);
+        } else {
+            return invocation.proceed();
```

Comment:
Do we need to throw exception or log warning message when m == null?

Reply:
When m=null,the method of original bean should be called.